backend/controllers/attendance/attendanceStatistics.js

```diff
@@ -326,9 +327,140 @@ const attendanceStatisticsFactory = (oauthApi, elite2Api, whereaboutsApi, logErr
     }
   }
 
+  const attendanceStatisticsSuspendedList = async (req, res) => {
+    const { agencyId, period, fromDate, toDate } = req.query || {}
+
+    const formattedFromDate = switchDateFormat(fromDate, 'DD/MM/YYYY')
+    const formattedToDate = toDate ? switchDateFormat(toDate, 'DD/MM/YYYY') : switchDateFormat(fromDate, 'DD/MM/YYYY')
+    const currentPeriod = getCurrentPeriod(moment().format())
+    const today = moment().format('DD/MM/YYYY')
+    const periods = period.split('_')
+
+    try {
+      const [user, caseloads, roles] = await Promise.all([
+        oauthApi.currentUser(res.locals),
+        elite2Api.userCaseLoads(res.locals),
+        oauthApi.userRoles(res.locals),
+      ])
+
+      const { activeCaseLoad, activeCaseLoadId } = extractCaseLoadInfo(caseloads)
+
+      if (!period || !fromDate) {
+        return res.redirect(
+          `${attendanceReasonStatsUrl}/suspended?agencyId=${activeCaseLoadId}&period=${currentPeriod}&fromDate=${today}&toDate=${today}`
+        )
+      }
+
+      const scheduledActivities = await elite2Api.getOffenderActivitiesOverDateRange(res.locals, {
+        agencyId,
+        fromDate: formattedFromDate,
+        toDate: formattedToDate,
+        period: period === 'AM_PM' ? '' : period,
+      })
+
+      const suspendedActivites = scheduledActivities.filter(activity => activity.suspended)
+      const totalOffenders = new Set(suspendedActivites.map(activity => activity.bookingId)).size
+
+      // Create mapping between date and a set of bookings with suspended activities on that date
+      // This mapping will be used to get the relevant attendances for each date
+      const suspendedBookingsPerEventDate = {}
+      suspendedActivites.forEach(activity => {
+        const date = moment(activity.startTime).format('YYYY-MM-DD')
+        if (!suspendedBookingsPerEventDate[date]) {
+          suspendedBookingsPerEventDate[date] = new Set()
+        }
+        suspendedBookingsPerEventDate[date].add(activity.bookingId)
+      })
+
+      const getAttendances = async (periodsList, data) => {
+        const suspendedAttendancesList = []
+        await Promise.all(
+          periodsList.map(async periodItem => {
+            await Promise.all(
+              Object.keys(data).map(async date => {
+                const attendances = await whereaboutsApi.getAttendanceForBookings(res.locals, {
+                  agencyId,
+                  period: periodItem,
+                  bookings: data[date],
+                  date,
+                })
+                suspendedAttendancesList.push(...attendances.attendances)
```

Comment:
This is a bit involved in order to accommodate the APIs current interface, which takes only one date and one period. Potential work to refactor the API endpoint to accommodate this use case will be done outside of this ticket.

Reply:
The maximum amount of calls to the whereabouts API from this page is 28 (14 days, 2 periods). This is not ideal, but is a rare use case where every single day would have suspended schedules in it.